src/proof/er/er_proof.cpp

```diff
@@ -0,0 +1,368 @@
+  for (const ErDefinition& def : d_definitions)
+  {
+    os << "\n    (clausify_rat_elimination_def _ _ _ "
+       << "er.def " << def.d_newVariable << " _ _ (\\ er.c" << firstDefClause
+       << " (\\ er.c" << (firstDefClause + 1) << " (\\ er.cnf"
+       << def.d_newVariable;
+
+    firstDefClause += 2 + def.d_otherLiterals.size();
+  }
+  parenCount += 4 * d_definitions.size();
+
+  // Unroll proofs of CNFs to proofs of clauses
+  firstDefClause = d_inputClauseIds.size() + 1;
+  for (const ErDefinition& def : d_definitions)
+  {
+    for (size_t i = 0, n = def.d_otherLiterals.size(); i < n; ++i)
+    {
+      os << "\n    (cnfc_unroll _ _ ";
+      os << "er.cnf" << def.d_newVariable;
+      if (i != 0)
+      {
+        os << ".u" << i;
+      }
+      os << " (\\ er.c" << (firstDefClause + 2 + i);
+      os << " (\\ er.cnf" << def.d_newVariable << ".u" << (i + 1);
+    }
+    parenCount += 3 * def.d_otherLiterals.size();
+
+    firstDefClause += 2 + def.d_otherLiterals.size();
+  }
+
+  // NB: At this point `firstDefClause` points to the first clause resulting
+  // from a resolution chain
+
+  // Now, elaborate each resolution chain
+  for (size_t cId = firstDefClause, nLines = d_tracecheck.d_lines.size();
+       cId <= nLines;
+       ++cId)
+  {
+    const std::vector<TraceCheckIdx>& chain =
+        d_tracecheck.d_lines[cId - 1].d_chain;
+    const std::vector<prop::SatLiteral> pivots = computePivotsForChain(chain);
+    Assert(chain.size() > 0);
+    Assert(chain.size() == pivots.size() + 1);
+
+    os << "\n    (satlem_simplify _ _ _ ";
+    parenCount += 1;
+
+    // Print resolution openings (reverse order)
+    for (int64_t i = pivots.size() - 1; i >= 0; --i)
+    {
+      prop::SatLiteral pivot = pivots[i];
+      os << "(" << (pivot.isNegated() ? 'Q' : 'R') << " _ _ ";
+    }
+
+    // Print resolution start
+    writeIdForClauseProof(os, chain[0]);
+    os << " ";
+
+    // Print resolution closings (forward order)
+    for (size_t i = 0, n = pivots.size(); i < n; ++i)
+    {
+      prop::SatVariable pivotVar = pivots[i].getSatVariable();
+      TraceCheckIdx clauseId = chain[i + 1];
+      writeIdForClauseProof(os, clauseId);
+      os << " ";
+      if (newVariables.find(pivotVar) != newVariables.end())
+      {
+        // This is a defined variable
+        os << "er.v" << pivotVar;
+      }
+      else
+      {
+        os << ProofManager::getVarName(pivotVar, "bb");
+      }
+      os << ") ";
+    }
+    os << "(\\ er.c" << cId;
+    parenCount += 1;
+  }
+
+  // Write proof of bottom
+  Assert(d_tracecheck.d_lines.back().d_clause.size() == 0,
+         "The TRACECHECK proof from drat2er did not prove bottom.");
+  os << "\n      er.c" << d_tracecheck.d_lines.back().d_idx
+     << " ; (holds cln)\n";
+
+  // Finally, close the parentheses!
+  std::fill_n(std::ostream_iterator<char>(os), parenCount, ')');
+}
+
+namespace {
+/**
+ * Resolves two clauses
+ *
+ * @param dest one of the inputs, and the output too
```

Comment:
So this is input and output param? The result will be stored in `dest`?

Reply:
Yes. I'll add a note to make this more clear.